index.ts

```diff
@@ -46,7 +46,7 @@ export class HttpSigning {
                 let value = config.headers?.hasOwnProperty(item) ? config.headers[item] : null;
 
                 if (item === '(request-target)') {
-                    value = config.method?.toLowerCase() + ' ' + (new URL(config.url as string)).pathname;
+                    value = config.method?.toLowerCase() + ' ' + config.url;
```

Comment:
```suggestion
                    const url = new URL(config.url as string);
                    value = config.method?.toLowerCase() + ' ' + url.pathname + url.search;
```

config.url can be a complete URL too, so we still need to parse to extract the path + query